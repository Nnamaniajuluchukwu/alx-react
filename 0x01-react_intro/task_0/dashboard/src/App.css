:root {
  --font-family-general: Arial, Helvetica, sans-serif;
}

.App {
  height: 100vh;
  max-width: 100vw;
}

.App-header {
  border-bottom: 3px solid #e0354b;
  color: #e0354b;
  margin: auto;
  display: flex;
  font-family: var(--font-family-general);
}

.App-header img {
  width: 200px;
  height: 200px;
}

.App-header h1 {
  position: relative;
  top: 4rem;
}

.App-body {
  border-bottom: 3px solid #e0354b;
  font-size: 1rem;
  font-family: var(--font-family-general);
  margin: auto;
  height: 23rem;
  padding: 3rem;
}

.App-footer {
  text-align: center;
  padding: 1rem;
  font-style: italic;
  font-family: var(--font-family-general);
}
