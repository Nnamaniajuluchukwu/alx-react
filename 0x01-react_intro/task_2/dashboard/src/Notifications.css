.Notifications {
    border: 3px dashed #e0354b;
    padding: 1rem;
}

li[data-priority='default'] {
    color: blue;
}

li[data-priority='urgent'], li {
    color: red;
}
