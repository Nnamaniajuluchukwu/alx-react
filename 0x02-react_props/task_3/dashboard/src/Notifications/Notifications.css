.Notifications {
    border: 3px dashed #e0354b;
    padding: 1rem;
}

li[data-notification-type='default'] {
    color: blue;
}

li[data-notification-type='urgent'], li {
    color: red;
}
