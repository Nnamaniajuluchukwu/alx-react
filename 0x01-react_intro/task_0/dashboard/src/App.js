import React from 'react';
import './App.css';

export default function App() {
  return (
    <div>
      <header className="App-header">
        <img src={require('./holberton_logo.jpg')} alt='holberton logo' />
        <h1>School dashboard</h1>
      </header>
      <body className="App-body">
        <p>Login to access the full dashboard</p>
      </body>
      <footer className='App-footer'>Copyright 2020 - holberton School</footer>
    </div>
  )
}
