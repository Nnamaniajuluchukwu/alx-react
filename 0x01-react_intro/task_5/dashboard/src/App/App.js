import React from 'react';
import './App.css';
import { getFullYear, getFooterCopy } from '../utils/utils';

export default function App() {
  return (
    <div>
      <header className="App-header">
        <img src={require('../assets/holberton_logo.jpg')} alt='holberton logo' />
        <h1>School dashboard</h1>
      </header>
      <body className="App-body">
        <p>Login to access the full dashboard</p>
        <label for='email'>Email </label>
        <input type='email' id='email'></input>
        <label for='password'> Password </label>
        <input type='password' id='password'></input>
        <button type='submit'>OK</button>
      </body>
      <footer className='App-footer'>Copyright {getFullYear()} - {getFooterCopy()}</footer>
    </div>
  )
}
