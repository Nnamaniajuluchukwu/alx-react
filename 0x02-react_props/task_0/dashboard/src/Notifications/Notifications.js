import React from "react";
import './Notifications.css';
import { getLatestNotification } from "../utils/utils";


export default function Notifications() {
    return (
        <div className="Notifications">
            <p>Here is the list of notifications</p>
            <ul>
                <li data-priority="default">New course available</li>
                <li data-priority="urgent">New resume available</li>
                <li dangerouslySetInnerHTML={{ __html: getLatestNotification() }}></li>
            </ul>
            <button style={{
                float: 'right',
                marginTop: '-9.5em',
                marginRight: '-0.5em',
                backgroundColor: '#fff',
            }} aria-label='Close' onClick={() => console.log('Close button has been clicked')}>
                <img src={require('../assets/close-button.png')} height="15px" width="13px" alt="close icon"/>
            </button>
        </div>
    );
}
