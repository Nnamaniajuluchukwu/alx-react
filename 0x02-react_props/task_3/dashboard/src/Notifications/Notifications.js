import React from "react";
import './Notifications.css';
import { getLatestNotification } from "../utils/utils";
import NotificationItem from "./NotificationItem";


export default function Notifications() {
    return (
        <div className='Notifications'>
            <p>Here is the list of notifications</p>
            <ul>
                <NotificationItem type='default' value='New course available' />
                <NotificationItem type='urgent' value='New resume available' />
                <NotificationItem html={{ __html: getLatestNotification() }} />
            </ul>
            <button style={{
                float: 'right',
                marginTop: '-9.5em',
                marginRight: '-0.5em',
                backgroundColor: '#fff',
            }} aria-label='Close' onClick={() => console.log('Close button has been clicked')}>
                <img src={require('../assets/close-button.png')} height="15px" width="13px" alt="close icon"/>
            </button>
        </div>
    );
}
