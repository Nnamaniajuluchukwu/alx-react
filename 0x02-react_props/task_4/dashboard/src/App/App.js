import React from 'react';
import './App.css';
import PropTypes from 'prop-types';
import CourseList from '../CourseList/CourseList';
import Login from '../Login/Login';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import Notifications from '../Notifications/Notifications';


export default function App({ isLoggedIn=false }) {
  return (
    <>
      <Notifications />
      <div className='App'>
        <Header />
        {isLoggedIn ? <CourseList /> : <Login />}
        <Footer />
      </div>
    </>
  )
}

App.propTypes = {
  isLoggedIn: PropTypes.bool
}
