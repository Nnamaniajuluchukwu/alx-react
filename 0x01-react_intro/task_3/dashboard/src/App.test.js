import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';
import Notifications from './Notifications';


describe('App Component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.exists()).toBeTruthy();
    });

    it('should renders a div with the class App-header', () => {
        const wrapper = shallow(<App />);
        const header = wrapper.find('.App-header');
        expect(header.exists()).toBeTruthy();
    });

    it('should renders a div with the class App-body', () => {
        const wrapper = shallow(<App />);
        const body = wrapper.find('.App-body');
        expect(body.exists()).toBeTruthy();
    });

    it('should renders a div with the class App-footer', () => {
        const wrapper = shallow(<App />);
        const footer = wrapper.find('.App-footer');
        expect(footer.exists()).toBeTruthy();
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(<App />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});


describe('Notifications Component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<Notifications />);
        expect(wrapper.exists()).toBeTruthy();
    });

    it('renders three list items', () => {
        const wrapper = shallow(<Notifications />);
        const listItems = wrapper.find('li');
        expect(listItems.length).toEqual(3);
    });

    it("renders the text 'Here is the list of notifications'", () => {
        const wrapper = shallow(<Notifications />);
        const text = wrapper.find('p');
        expect(text.props().children).toEqual('Here is the list of notifications');
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(<Notifications />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
