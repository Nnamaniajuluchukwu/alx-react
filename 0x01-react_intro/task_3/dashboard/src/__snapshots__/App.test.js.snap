// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match the snapshot 1`] = `
<div>
  <header
    className="App-header"
  >
    <img
      alt="holberton logo"
      src="holberton_logo.jpg"
    />
    <h1>
      School dashboard
    </h1>
  </header>
  <body
    className="App-body"
  >
    <p>
      Login to access the full dashboard
    </p>
    <label
      for="email"
    >
      Email 
    </label>
    <input
      id="email"
      type="email"
    />
    <label
      for="password"
    >
       Password 
    </label>
    <input
      id="password"
      type="password"
    />
    <button
      type="submit"
    >
      OK
    </button>
  </body>
  <footer
    className="App-footer"
  >
    Copyright 
    2023
     - 
    Holberton School main dashboard
  </footer>
</div>
`;

exports[`Notifications Component should match the snapshot 1`] = `
<div
  className="Notifications"
>
  <p>
    Here is the list of notifications
  </p>
  <ul>
    <li
      data-priority="default"
    >
      New course available
    </li>
    <li
      data-priority="urgent"
    >
      New resume available
    </li>
    <li
      dangerouslySetInnerHTML={
        Object {
          "__html": "<strong>Urgent requirement</strong> - complete by EOD",
        }
      }
    />
  </ul>
  <button
    aria-label="Close"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "float": "right",
        "marginRight": "-0.5em",
        "marginTop": "-9.5em",
      }
    }
  >
    <img
      alt="close icon"
      height="15px"
      src="close-button.png"
      width="13px"
    />
  </button>
</div>
`;
